### DISCUSSION TITLE
Push job with TTL

### DISCUSSION THREAD

**Discussion by vbocan** [apalis-dev/apalis#46] [Ideas]
Is it possible to add TTL to a job? My scenarios implies a worker queue with items of several MB in size which I would like to expire after a while, to save on disk space.

**Comment by geofmureithi** [comment]
It may be possible. What are you using for storage?
The approach would be writing a [Layer](https://tower-rs.github.io/tower/tower/trait.Layer.html), something like `TtlLayer`.
Would this fix your problem?

**Reply by vbocan** [reply]
I am using Redis for storage. The TtlLayer approach may be what it takes, but I am not familiar with Tower.

**Reply by geofmureithi** [reply]
This might be a good start: https://docs.rs/tower/latest/tower/timeout/struct.TimeoutLayer.html
